File: src/utils/dateUtils.ts
export function getLocalDateString(date: Date = new Date()): string {
  const y = date.getFullYear();
  const m = String(date.getMonth() + 1).padStart(2, '0');
  const d = String(date.getDate()).padStart(2, '0');
  return `${y}-${m}-${d}`;
}

export function isWeekend(date: Date = new Date()): boolean {
  const day = date.getDay();
  return day === 0 || day === 6;
}

export function getBudgetForDate(
  category: { weekday_budget_min: number; weekend_budget_min: number },
  date: Date = new Date()
): number {
  return isWeekend(date)
    ? category.weekend_budget_min
    : category.weekday_budget_min;
}

export function formatDuration(totalSec: number): string {
  const h = Math.floor(totalSec / 3600);
  const m = Math.floor((totalSec % 3600) / 60);
  const s = totalSec % 60;
  if (h > 0) return `${h}:${String(m).padStart(2, '0')}:${String(s).padStart(2, '0')}`;
  return `${m}:${String(s).padStart(2, '0')}`;
}

export function formatTime(unixSec: number): string {
  const d = new Date(unixSec * 1000);
  const h = String(d.getHours()).padStart(2, '0');
  const m = String(d.getMinutes()).padStart(2, '0');
  return `${h}:${m}`;
}

File: lib/utils/date_utils.dart
// lib/utils/date_utils.dart

/// ローカル日付を YYYY-MM-DD 形式で返す
String getLocalDateString([DateTime? dt]) {
  final d = dt ?? DateTime.now();
  return '${d.year}-${d.month.toString().padLeft(2, '0')}-${d.day.toString().padLeft(2, '0')}';
}

/// 土・日を休日と判定
bool isWeekend([DateTime? dt]) {
  final d = dt ?? DateTime.now();
  return d.weekday == DateTime.saturday || d.weekday == DateTime.sunday;
}

/// 指定日の予算分数を返す（平日 or 休日）
int getBudgetForDate(int weekdayBudgetMin, int weekendBudgetMin, [DateTime? dt]) {
  return isWeekend(dt) ? weekendBudgetMin : weekdayBudgetMin;
}

/// 経過秒数を HH:MM:SS 形式にフォーマット
String formatDuration(int totalSec) {
  final h = totalSec ~/ 3600;
  final m = (totalSec % 3600) ~/ 60;
  final s = totalSec % 60;
  return '${h.toString().padLeft(2, '0')}:${m.toString().padLeft(2, '0')}:${s.toString().padLeft(2, '0')}';
}

/// 曜日の日本語表記
const _weekdayJa = ['月', '火', '水', '木', '金', '土', '日'];
String weekdayJa([DateTime? dt]) {
  final d = dt ?? DateTime.now();
  return _weekdayJa[d.weekday - 1];
}
